docs/guides/04-data-driven-visualizations-part-2.md

```diff
@@ -0,0 +1,113 @@
+## Data-driven visualizations (part 2)
+
+### Ramp values
+
+In the previous section we talked about using different types of input for ramp, but we always output colors picked from a list. `ramp` supports to use other types of outputs and also CARTO VL includes some fixed constant palettes of colors. Let's see it!
+
+#### Color values
+
+One way to output colors is to specify a list of colors, just like we have done in all the previous examples. This can be done with expressions like `ramp($dn, [blue, red])`. But usage of named colors (`blue`, `red`, `green`...) is not enforced, any valid color expression is ok, for example:
+- `ramp($dn, [rgb(200,220,222), rgba(200,120,22, 0.8)])`
+- `ramp($dn, [hsv(0,1,1), hsv(0.5,1,1)])`,`ramp($dn, [#00F, #F00])`
+- `ramp($dn, [blue, #F00])`
+- `ramp($dn, [opacity(blue, 0.4), opacity( #F00, 0.6),])`
+
+There is also another way to specify colors, and that is to use one of the built-in color palettes. We have built-in all the CARTOColors and ColorBrewer palettes. You can use them like this:
+- `ramp($dn, temps)`
+- `ramp($dn, tealrose)`
+- `ramp($dn, cb_blues)`
+
+The complete list of CARTOColors can be seen [here](https://carto.com/carto-colors/).
+
+Let's see all these options in actions!
+
+<div class="example-map">
+    <iframe
+        id="population-density-colors"
+        src="/developers/carto-vl/examples/maps/guides/ramp/population-density-colors.html"
```

Comment:
Very nice example. As in guide-part-1, we should try to adjust a bit the titles' style

Reply:
I don't understand the meaning of `Legend of e($dn, e($dn), e($dn))`

Reply:
That seems like a regression on the implementation of legend, not a bug on the example, I'll take a look though

Reply:
https://github.com/CartoDB/carto-vl/issues/1059

Reply:
That has been already solved and merged